File: statistics/python/weigth_variance.py
# -*- coding: utf-8 -*-
# @Author: Admin
# @Date:   2020-01-10 01:44:06
# @Last Modified by:   Admin
# @Last Modified time: 2020-01-10 01:49:44
def weigthVariance(numberData=[],weigth=[]):
    totalWigth,totalNumebr = 0,0
    if numberData.__len__()== weigth.__len__():
        i=0
        while i<=len(numberData):
            totalWigth =totalWigth+weigth[i]
            totalNumebr = totalNumebr+numberData[i]*weigth[i]
            i=i+1
    return totalNumebr/totalWigth

File: statistics/weigth_variance.ts
/**
 * 加权平均数
 */
    /**
     * 计算加权平均数
     * @param {Array<number>} numberData 要计算的值
     * @param {Array<number>} weigth 权值
     */
    function weigthVariance([...numberData]: Array < number > , [...weigth]: Array < number > ) {
    	let totalWigth:number=0,totaNumber:number=0;
        if (numberData.length == weigth.length) {
        	for(let i:number =0;i<=numberData.length;i++){
        		totalWigth+=weigth[i];
        		totaNumber+=numberData[i]*weigth[i];
        	}
        }else{
        	throw Error("权重值和数据不匹配!");
        }
        return totaNumber/totalWigth;
    }
export {weigthVariance};